queasars/minimum_eigensolvers/evqe/evqe.py

```diff
@@ -0,0 +1,205 @@
+        purpose, the sampler is used instead. If reproducible behaviour is required, the seed option of the estimator
+        needs to be set
+    :type estimator: Optional[Estimator]
+    :param sampler: Sampler primitive used to measure the circuits QuasiDistribution. If reproducible behaviour is
+        required, the seed option of the estimator needs to be set
+    :type sampler: Sampler
+    :param optimizer: Qiskit optimizer used to optimize the parameter values. Should be configured to terminate after
+        a relatively low amount of circuit evaluations to enable the gradual evolution of the individuals
+    :type optimizer: Optimizer
+    :param optimizer_n_circuit_evaluations: amount of circuit evaluations expected per optimizer run, None if unknown
+    :type optimizer_n_circuit_evaluations: Optional[int]
+    :param dask_client: used for parallelization of the evolutionary algorithm. If no dask client is given, a
+        LocalCluster using multiprocessing is used. If reproducible behaviour is required, the cluster should use
+        multiprocessing with only one worker per process
+    :type dask_client: Client
+    :param max_generations: Maximum amount of generations the evolution may go on for. Either max_generations or
+        max_circuit_evaluations or termination_criterion needs to be provided
+    :type max_generations: Optional[int]
+    :param max_circuit_evaluations: Maximum amount of circuit evaluations the solver may use. Depending
+        on the configuration this measure may be undershot or overshot significantly. Either max_generations or
+        max_circuit_evaluations or termination_criterion needs to be provided
+    :type max_circuit_evaluations: Optional[int]
+    :param termination_criterion: criterion which defines how to determine whether the solver has converged.
+        Either max_generations or max_circuit_evaluations or termination_criterion needs to be provided
+    :type termination_criterion: Optional[EvolvingAnsatzMinimumEigensolverBaseTerminationCriterion]
+    :param random_seed: Optional integer value to control randomness. For this to be effective the seed option for
+        the sampler (and if applicable the estimator) need to be set and the dask client must use one separate process
+        per worker
+    :type random_seed: Optional[int]
+    :param population_size: amount of individuals within a population
+    :type population_size: int
+    :param randomize_initial_population_parameters: Determines whether the parameter values of the individuals in
+        the first population shall be initialized randomly or at 0. If during the optimization process no improvement
+        occurs at all this may be due to the optimizer not being able to detect a gradient at parameter values of 0
+    :type randomize_initial_population_parameters: bool
+    :param speciation_genetic_distance_threshold: Genetic distance (amount of circuit layers not shared
```

Comment:
The speciation operator is currently EVQE specific, which is why it only uses the genetic distance measure proposed in their paper.  Would you rather want a generic speciation operator? This would neccessitate the generic population type to model speciation as well as generic types for both the genetic distance and the clustering into species.

Reply:
As discussed yesterday, it's fine since this is an EVQE implementation as in the paper. If we later want to play around with other genetic distances we can still add this.